### DISCUSSION TITLE
How to synchronize waveform and spectrogram

### DISCUSSION THREAD

**Discussion by srdecny** [katspaugh/wavesurfer.js#3535] [Q&A]
I've noticed that the spectrogram is ever so slightly shifted compared to the waveform. Is there any way to make them perfectly in sync? I understand that the waveform is computed using a sliding window algorithm, so there's some natural lookahead/lookbehind. Not sure if this isn't a bug, actually. The screenshots are using the `examples/audio/ahoy.mp4` file, which is quite short, so it shouldn't be related to #2737.

<img width="438" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/7964806/6b027efd-09c2-45e3-89e1-4766aa358c93">
<img width="414" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/7964806/f20ed61c-4db2-46e9-9a50-85690e77bea8">

**Comment by katspaugh** [comment]
Looks like a bug indeed!
